### ISSUE TITLE
Game Crash

### ISSUE THREAD

**Issue by chilpol** [GregRos/IEMod.pw#90]
Game crashing constantly while playing the game with the mod. Loading the same save without mod played fine. What information do you need to investigate this? :)

**Comment by oddrheia** [comment]
Likely the culprit savegame file, before the fight that crashes. I know  Llere on the nexus page of the mod have been asking people for the saves with reproducible crashes, so try sending them a private mesage  and see if you can mail them the file or two.

**Comment by llerelol** [comment]
yea, i'm looking into it. no promises tho - it takes much more time than i thought it would
if you wanna help, please share save games or info about the moment it crashed.

**Comment by chilpol** [comment]
I have the entire folder zip up. I do not know which file to send you exactly :)
https://www.dropbox.com/s/6t22wmc7uvxqm4r/Pillars%20of%20Eternity.rar?dl=0

**Comment by SonicZentropy** [comment]
Not sure how helpful this will be, but I've narrowed my crashes down quite a bit.  Something in the IEMod framework code is causing them.  I tore out all of the discrete "mods", leaving only the command line, UI, Options, and Helpers stuff -- the crashes continued every time I loaded my test saves.  Then I ripped all of that out and put some of the mods back in by themselves, hard coded their options so the DLL would compile without needing IEOptions and such, and rebuilt the DLL to use those naked mods only...crashes vanished.

I called it done there since it's frustratingly close to impossible to debug effectively at all.  I also don't really have any idea how to turn this info into an actual solution for people who don't want to/know how to gut and rebuild their own DLL, but it might save somebody else time debugging!

**Comment by llerelol** [comment]
yeah, i came to exactly same conclusions, i also traced down the function that makes game crash which is AIState:BaseUpdate() and i have no idea what can be done to fix this.

**Comment by GregRos** [comment]
@llerelol Can't you just remove the mod that requires this function? I think it is FastSneak.

Just exclude the whole `FastSneak` folder from the project. Or else use put `[assembly: DisablePatchingByName({the namespace})]` somewhere outside of a class.

If the source is actually [mod_PartMemberAI](https://github.com/IEMod/IEMod.pw/blob/25897b8838f6546dcf0796ac33a7649833cbe5d3/IEMod/Mods/FastSneak/mod_PartyMemberAI.cs) which is translated to an `AIState:Update` call.

Also, there in one of the folders, called `debug_log` or something like that. It should contain info about why the game crashed.

**Comment by llerelol** [comment]
Its not that simple, i use the code from fastsneak mod as a standalone module for patchwork and it works flawless so its not about it. Something else is interfering, as SonicZentropy said even if you disable all the mods and only leave like core code it still causes crashes. I've looked through all of it and i couldnt find a problem. I kind of given up at that point, spent too much time debugging it and got no results.

**Comment by Thormind** [comment]
Would you be able to give us that fast sneak module as a standalone mod, or explain how it can be done please?  Many players are using ie mod just for that part.  Would probably be easier and faster to update when new patches come out...

**Comment by llerelol** [comment]
code is in my fork, binary is here - https://drive.google.com/open?id=0B0mskSi9O7aGVnhTVEZSUUlrdGM, i've posted it on nexus a while ago.

**Comment by Thormind** [comment]
Thank you sooo much!

**Comment by Thormind** [comment]
Is there a way to apply the sneak module without using the iemod launcher? The game still crashes during combat even if i only enable the fast sneak module you just linked.

**Comment by llerelol** [comment]
no way, i've completed a playthrough myself with fastsneak and lootshuffler modules and game hasnt crashed a single time. it was on patch 3.01 tho but none of the code changed, so idk.

**Comment by llerelol** [comment]
ok ill recompile those modules using latest assembly, no idea if that helps but still.
and yea make sure you delete iemod.pw if youre using standalone modules, because if you just untick it it still loads.

Update: ok i reuploaded files, same link. i still highly doubt that was a problem.

**Comment by Thormind** [comment]
I made copy of my managed folder before intalling iemod.  I deleted the modified folder and pasted my backup.  I also removed the iemod files from the launcher folder, like you suggested.  No crash since then.  tks!

**Comment by Thormind** [comment]
One last question.  In iemod theres an option to "superspeed" the game by pressing ctrl+fast key.  Is there a way to include that option in the faststealth module?

**Comment by HK47196** [comment]
I get a reproducible crash during any long fights, is there any way to get a backtrace on crash so I can help debug it? For now when it happens I just launch it without iemod via patchwork, finish the fight, then switch back to using iemod.

I'm on Linux if that makes a difference.
